dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/DeletedUsernameDAO.java

```diff
@@ -0,0 +1,37 @@
+package io.dockstore.webservice.jdbi;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+import io.dockstore.webservice.core.DeletedUsername;
+import io.dropwizard.hibernate.AbstractDAO;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+
+public class DeletedUsernameDAO extends AbstractDAO<DeletedUsername> {
+    public DeletedUsernameDAO(SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    public DeletedUsername findById(Long id) {
+        return get(id);
+    }
+
+    public long create(DeletedUsername deletedUsername) {
+        return persist(deletedUsername).getId();
+    }
+
+    public void delete(DeletedUsername deletedUsername) {
+        Session session = currentSession();
+        session.delete(deletedUsername);
+        session.flush();
+    }
+
+    public List<DeletedUsername> findByUsername(String username) {
+        return list((this.currentSession().getNamedQuery("io.dockstore.webservice.core.DeletedUsername.findByUsername").setParameter("username", username)));
+    }
+
+    public DeletedUsername findNonReusableUsername(String username, Timestamp timestamp) {
+        return uniqueResult(this.currentSession().getNamedQuery("io.dockstore.webservice.core.DeletedUsername.findNonReusableUsername").setParameter("username", username).setParameter("timestamp", timestamp));
```

Comment:
What happens if there is more than one result? Does this fail or return the first result?

It would take at least 3 years for it to happen, but it is possible in theory.

Reply:
It will fail if there's more than one result. But I think it should only ever give one at result at a time. There should only be one unusable entry for a username at a given time since other users can't change or a create an account with that username for 3 years. Once the 3 years is up, then it won't be found by the query.

Reply:
I still think it's possible after 6 years, although **extremely** unlikely, because, unless I'm missing it, you're not deleting the users from the deletedusers table.

1. jdoe account is deleted, it goes into deleted_users table?
2. 3 years later jdoe account is used again
3. jdoe account is deleted again -- won't that put a second entry in the deleted_users table?

Reply:
Yeah there would be a second entry in the deleted_users table, but the query itself is only looking for deleted usernames that are less than 3 years old. 

`SELECT u FROM DeletedUsername u WHERE u.username = :username AND u.dbCreateDate > :timestamp`

So only the second jdoe account would show up in the query since the first one is now older than 3 years.